ext/opcache/shared_alloc_mmap.c

```diff
@@ -67,14 +69,15 @@ static void *find_prefered_mmap_base(size_t requested_size)
 	while (fgets(buffer, MAXPATHLEN, f) && sscanf(buffer, "%lx-%lx", &start, &end) == 2) {
 		if ((uintptr_t)execute_ex >= start) {
 			/* the current segment lays before PHP .text segment or PHP .text segment itself */
-			if (last_free_addr + requested_size <= start) {
-				last_candidate = last_free_addr;
+			if (start - pre_seg_end >= requested_size + huge_page_size) {
+				last_candidate = start - requested_size - huge_page_size;
+				last_candidate = ZEND_MM_ALIGNED_SIZE_EX(last_candidate, huge_page_size);
```

Comment:
This code may make a single huge page "hole" between the ``.text`` segment and the SHM.
Can you check the aligned addresses to avoid this?

Reply:
Thank you for your comment. 
An extra hugepage size space is reserved here to prevent the size of the reserved space from being insufficient when the address is aligned. The memory space is large enough, so the problem of this single huge page "hole" was not considered before. 
Maybe additional judgment ("if") is required here to resolve it.

Reply:
Thanks, done.